src/Conversion/ONNXToKrnl/Math/Elementwise.cpp

```diff
@@ -331,6 +334,77 @@ struct ScalarOp<ONNXTanOp> {
   using IOp = NotSuportedScalarOp;
 };
 
+//===----------------------------------------------------------------------===//
+// Scalar unary ops for lowering ONNXGeluOp
+//===----------------------------------------------------------------------===//
+template <>
+struct ScalarOp<ONNXGeluOp> {
+  using FOp = CustomScalarOp;
+  using IOp = CustomScalarOp;
+};
+
+template <>
+double analyzeSimdFor<ONNXGeluOp>(
+    Type t, Operation *op, int64_t &von, int64_t &son) {
+  double results;
+  StringRef approximate = dyn_cast<ONNXGeluOp>(op).getApproximate();
+  if (approximate.equals_insensitive("none"))
+    results = simdAnalysis(
+        {GenericOps::ArithmeticGop, GenericOps::MulGop}, {1, 3}, t, von, son);
+  if (approximate.equals_insensitive("tanh"))
+    results = simdAnalysis(
+        {GenericOps::ArithmeticGop, GenericOps::MulGop, GenericOps::PowGop},
+        {2, 4, 1}, t, von, son);
+  return results;
```

Comment:
Since pow(3) is really 2 matmul, I would remove the Pow and add 1 to MulGop. Note that we also have `TrigHyperbolicGop` so it can be added too.

Reply:
Done!